test/parallel_api/numeric/numeric.ops/exclusive_scan.pass.cpp

```diff
@@ -0,0 +1,102 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(numeric)
+
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+#include "support/sycl_alloc_utils.h"
+
+template <sycl::usm::alloc alloc_type>
+void
+test_with_usm(sycl::queue& q, const ::std::size_t __count)
+{
+    auto prepare_data = [__count](std::vector<::std::size_t>& idx, std::vector<int>& val)
+    {
+        for (int i = 0; i < __count; i++)
+        {
+            idx[i] = i + 1;
+            val[i] = 0;
```

Comment:
I think we should explain why we fill an output sequence by zero.

Reply:
Fixed